tez-api/src/main/java/org/apache/tez/runtime/api/events/InputDataInformationEvent.java

```diff
@@ -79,6 +79,12 @@ public static InputDataInformationEvent createWithObjectPayload(int srcIndex,
     return new InputDataInformationEvent(srcIndex, userPayloadDeserialized, null);
   }
 
+  public static InputDataInformationEvent createWithSerializedPath(int srcIndex, String serializedPath) {
+    InputDataInformationEvent event = new InputDataInformationEvent(srcIndex, null);
+    event.serializedPath = serializedPath;
```

Comment:
Won't just be easier to add a 3rd constructor for the InputDataInformationEvent? To set a property is fine, just not fail safe.

Reply:
this method makes it impossible for users to call constructors in an invalid way (like: adding payload and serialized path at the same time), also with having a method name that implies this behavior, what constructor are you suggesting that has the same advantages?

Reply:
ack